enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -281,36 +275,44 @@ func (s *SociArtifactStore) GetArtifacts(ctx context.Context, req *socipb.GetArt
 	return &resp, nil
 }
 
-func blobKey(hash string) string {
-	return blobKeyPrefix + hash
+// Accepts a container image config hash 'h' which uniquely identifies the
+// image and generates the repb.Digest that should be used as the cache key
+// for storing the SOCI Index for that image.
+func sociIndexKey(h v1.Hash) (*repb.Digest, error) {
+	buf := bytes.NewBuffer([]byte(h.Hex + *sociIndexCacheSeed))
+	return digest.Compute(buf, repb.DigestFunction_SHA256)
 }
 
-func resourceName(digest *repb.Digest) *rspb.ResourceName {
+func resourceName(digest *repb.Digest, cacheType rspb.CacheType) *rspb.ResourceName {
 	return &rspb.ResourceName{
 		Digest:         digest,
 		Compressor:     repb.Compressor_IDENTITY,
-		CacheType:      rspb.CacheType_CAS,
+		CacheType:      cacheType,
 		DigestFunction: repb.DigestFunction_SHA256,
 	}
 }
```

Comment:
could we use digest.NewResourceName directly here instead?

func NewResourceName(d *repb.Digest, instanceName string, cacheType rspb.CacheType, digestFunction repb.DigestFunction_Value) *ResourceName {...}

That way we'd be less likely to end up with a partially-filled-out rspb.ResourceName proto.

Even if you just called it from this method, would be an improvement:

```
func resourceName(digest *repb.Digest, cacheType rspb.CacheType) *rspb.ResourceName {
  return digest.NewResourceName(d, "" /*instnaceName*/, cacheType, repb.DigestFunction_SHA256).ToProto()
}
```

Reply:
Oh shoot, forgot about that, thanks